### PR TITLE
Generate SDK version into a source file using a velocity template

### PR DESCRIPTION
**Pull request by kiiadi**
The SDK version is known at build time; this saves having to load and parse a class at runtime.

### CODE REVIEW

File: core/src/test/java/software/amazon/awssdk/util/VersionInfoTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.Test;
+
+public final class VersionInfoTest {
+
+    @Test
+    public void versionIsTheSameAsMavenProject() throws Exception {
+        assertThat(VersionInfo.SDK_VERSION).isEqualTo(getSdkVersionFromPom());
+    }
+
+    private String getSdkVersionFromPom() throws URISyntaxException, IOException {
```
Comment (shorea) [inline core/src/test/java/software/amazon/awssdk/util/VersionInfoTest.java]: eww, can we just set a system property that we read?

Comment (kiiadi) [inline core/src/test/java/software/amazon/awssdk/util/VersionInfoTest.java]: you don't like my awesome pom loader?

Comment (sarxos) [inline core/src/test/java/software/amazon/awssdk/util/VersionInfoTest.java]: it seems like overkill, why not simply https://maven.apache.org/plugins/maven-resources-plugin/examples/filter.html

Comment (kiiadi) [inline core/src/test/java/software/amazon/awssdk/util/VersionInfoTest.java]: @sarxos happy to go take another look, but this generates an actual class that (I assume) won't be compiled if it's considered to be a resource by maven, it'll just get copied into the target/classes directory as is.

**Pull request merged**